pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -353,6 +354,14 @@ func generateRawContent(object *unstructured.Unstructured) ([]byte, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to get the ports field in Serivce object, name =%s: %w", object.GetName(), err)
 		}
+	} else if object.GetKind() == "Job" && object.GetAPIVersion() == batchv1.SchemeGroupVersion.String() {
+		if manualSelector, exist, _ := unstructured.NestedString(object.Object, "spec", "manualSelector"); !exist || manualSelector == "true" {
```

Comment:
Hi Ryan! This should be `!exist || manualSelector == "false"`, right? If the `manualSelector` is `true`, the customer has its own `.spec.selector` specified and we shouldn't clear the value.